src/cmd/ksh93/sh/array.c

```diff
@@ -132,17 +132,22 @@ static int arsize(struct index_array *ap, int maxi) {
 
 static struct index_array *array_grow(Namval_t *, struct index_array *, int);
 
-// Return index of highest element of an array.
+// Return next index after the highest element in an array.
 int array_maxindex(Namval_t *np) {
     struct index_array *ap = (struct index_array *)nv_arrayptr(np);
     int i = ap->maxi;
     if (is_associative(ap)) return (-1);
```

Comment:
Mind removing the pointless parens around the return value since you're in the neighborhood? :smile: